packages/cli/src/utils/prereq-checks.ts

```diff
@@ -0,0 +1,244 @@
+    return true;
+  } else {
+    throw new Error(message);
+  }
+}
+
+// should cover all eslint extensions
+export function isESLintPreReq(basePath: string, requiredParser: string): boolean {
+  const lintConfigPath = getEsLintConfigPath(basePath);
+  if (!lintConfigPath) {
+    throw new Error(
+      `Eslint config (.eslintrc.{js,yml,json,yaml,cjs}) does not exist. Please add one and try again.`
+    );
+  }
+
+  // read the lintConfigPath as text and check if it has the requiredParser string in it
+  const lintConfigText = readFileSync(lintConfigPath, 'utf-8');
+
+  if (!lintConfigText.includes(requiredParser)) {
+    throw new Error(
+      `Please update the parser to "${requiredParser}" in the ESLint config and try again.`
+    );
+  }
+
+  return true;
+}
+
+export function getEsLintConfigPath(basePath: string): string {
+  // glob against the following file extension patterns and return the first match
+  const configPath = fastGlob.sync(join(basePath, '.eslintrc?(.{js,yml,json,yaml,cjs})'))[0];
+  return configPath;
+}
+
+function isTSConfigGlintPreReq(
+  tsConfigPath: string,
+  requiredTSConfigGlint: Pick<PreReqTSConfig, 'glint'>,
+  message?: string
+): void {
+  // the glint key isn't supported by with ts API so we have to grab it manually
+  const { glint } = readTSConfig<TSConfig>(tsConfigPath);
+  const { glint: glintRequired } = requiredTSConfigGlint;
+
+  if (!glint && glintRequired) {
+    throw new Error(`"glint" key and options missing in tsconfig.json. ${message}`);
+  }
+
+  if (glint && glintRequired) {
+    // check if the glint key has all the required keys
+    const missingKeys = Object.keys(glintRequired).filter(
+      (key) => !Object.keys(glint).includes(key)
+    );
+    if (missingKeys.length > 0) {
+      throw new Error(
+        `"glint" key is missing the following options in tsconfig.json: ${missingKeys.join(
+          ', '
+        )}. ${message}`
+      );
+    }
+
+    // this is default true, so if its explicitly set to false throw
+    if (glint.checkStandaloneTemplates === false) {
+      throw new Error(
+        `"glint.checkStandaloneTemplates" value is incorrect in tsconfig.json. ${message}`
+      );
+    }
+  }
+}
+
+export function isExistsESLintConfig(basePath: string): boolean {
+  const lintConfigPath = getEsLintConfigPath(basePath);
+  if (!lintConfigPath) {
+    throw new Error(`Eslint config (.eslintrc.{js,yml,json,yaml,cjs}) does not exist.`);
+  }
+  return true;
+}
+
+export function isExistsPackageJSON(basePath: string): boolean {
+  const packageJsonPath = resolve(basePath, 'package.json');
+  if (!existsSync(packageJsonPath)) {
+    throw new Error(
+      `${packageJsonPath} does not exists. Please run rehearsal inside a project with a valid package.json.`
+    );
+  }
+  return true;
+}
+
+export function isExistsTSConfig(basePath: string): boolean {
+  const tsConfigPath = resolve(basePath, 'tsconfig.json');
```

Comment:
I think this: https://github.com/rehearsal-js/rehearsal-js/blob/6363ff411e8b24cd1badd62bd4de3e437218a111/packages/migrate/src/migrate.ts#L69-L73

Reply:
https://github.com/rehearsal-js/rehearsal-js/issues/1056